web/src/compose_notifications.ts

```diff
@@ -291,16 +296,16 @@ export function notify_messages_outside_current_search(messages: Message[]): voi
             continue;
         }
         const above_composebox_narrow_url = get_above_composebox_narrow_url(message);
-        const link_text = $t(
-            {defaultMessage: "Narrow to {message_recipient}"},
-            {message_recipient: get_message_header(message)},
+        const link_content = $t_html(
+            {defaultMessage: "Narrow to <z-message-recipient></z-message-recipient>"},
+            {"z-message-recipient": () => get_message_header(message)},
         );
         notify_above_composebox(
             $t({defaultMessage: "Sent! Your message is outside your current view."}),
             compose_banner.CLASSNAMES.narrow_to_recipient,
             above_composebox_narrow_url,
             message.id,
-            link_text,
+            link_content,
```

Comment:
We have a naming convention that HTML objects being passed into templates need to be explicitly tagged as `html_`; let's do that with this and `channel_topic`.